packages/lpa-questionnaire-web-app/src/controllers/upload-plans.js

```diff
@@ -8,3 +9,80 @@ exports.getUploadPlans = (req, res) => {
     uploadedFiles: [],
   });
 };
+
+exports.postUploadPlans = async (req, res) => {
+  const documents = req.body?.files?.documents || [];
+
+  const { delete: deleteFile = '', errors = {}, submit = '' } = req.body;
+
+  // Chance for delete to be triggered due to non-JS solution. delete will be set to value of filename if button clicked
+  if (deleteFile) {
+    // TODO: add handling for deletion from DB and doc store as part of AS-1538
+    const file = req.session.uploadedFiles?.find((upload) => upload.name === deleteFile);
```

Comment:
I'm curious on this, can't seem to find an immediate answer.

As best I understand it, with the questionnaire the user can return to the site at a later date and resume - using an id in the URL. 

What happens with previously uploaded files in this instance?

I'm guessing it works because the session is long lived(?)

However, I'm also thinking we should be deleting the session when the user finally submits at the end. Now, weirdly, the code I added in to do this for FWA has been removed, which I'm guessing is because during dev / debug, it's a PITA, but that's a different issue. 

Just curious as to how that might play out? Maybe it's not even an issue.

Reply:
I'm with you, missing some context. The session is used far less than that - as soon as someone hits save and continue it writes to the doc store. If they leave the question and reopen, it clears unsaved docs by way of middleware. In this way it neatly avoids the issue.

The next ticket handles the upload and saving. Deletion of a document from that point will delete from the doc store.

Hope that makes sense?

Reply:
Yes, thanks for the explanation - much appreciated.